activesupport/lib/active_support/claims.rb

```diff
@@ -0,0 +1,63 @@
+require 'active_support/time'
+
+module ActiveSupport
+  # +Claims+ supports <tt>ActiveSupport::MessageVerifier</tt> by
+  # representation of message in the form JWT Claims and verifying
+  # the claims of the signed message. 
+  class Claims
+    class InvalidClaims < StandardError; end
+    class ExpiredClaims < StandardError; end
+    
+    attr_reader :payload, :purpose, :expires_at
+    DEFAULT_PURPOSE = 'universal'
+
+    # Keyword argument +:payload+ is mandatory
+    def initialize(payload:, **options)
+      @payload = payload
+      @purpose = options[:for] || DEFAULT_PURPOSE
+      @expires_at = pick_expiration(options)
+    end
+
+    class << self
+      attr_accessor :expires_in
+
+      def verify!(claims, options = {})
+        if claims[:for] == (options[:for] || DEFAULT_PURPOSE)
+          claims[:pld] if parse_expiration(claims[:exp])
+        else
+          raise InvalidClaims
+        end        
+      end
+
+      private
+        def parse_expiration(expiration)
+          return true unless expiration
+
+          Time.iso8601(expiration).tap do |timestamp|
+            raise ExpiredClaims if Time.now.utc > timestamp
+          end
+        end
+    end
+
+    def to_h
+      { pld: @payload, for: @purpose.to_s }.tap do |claims|
+        claims[:exp] = @expires_at.utc.iso8601(3) if @expires_at
+      end
+    end
+
+    def ==(other)
+      other.is_a?(self.class) && @purpose == other.purpose && @payload == other.payload
+    end
+
+    private
+      def pick_expiration(options)
+        return options[:expires_at] if options.key?(:expires_at)
+
+        # Expiration defaults to 1 month through class level
+        # variable +:expires_in+ set in railtie.
```

Comment:
This shouldn't be in a private method's definition. It would be better in the general Claims documentation.